tests/wsl/install/ms_win_installation.pm

```diff
@@ -26,53 +26,63 @@ sub prepare_win11 {
 }
 
 sub run {
+
+    my $self = shift;
+
     if (get_var('UEFI')) {
-        assert_screen 'windows-boot';
-        send_key 'spc';    # boot from CD or DVD
+        while (check_screen('windows-boot', timeout => 5)) {
+            send_key 'spc';
+            record_info('SPC', 'Space key pressed');
+        }    # boot from CD or DVD
     }
-    # This test works onlywith CDMODEL=ide-cd due to windows missing scsi drivers which are installed via scsi iso
-    assert_screen 'windows-setup', 300;
-    send_key 'alt-n';    # next
-    prepare_win11 if (check_var('WIN_VERSION', '11'));
 
-    save_screenshot;
-    send_key 'alt-i';    # install Now
-    save_screenshot;
-    send_key 'alt-n';    # next
-    assert_screen 'windows-activate';
-    if (my $key = get_var('_SECRET_WINDOWS_10_PRO_KEY')) {
-        type_password $key . "\n";
-        assert_screen([qw(windows-wrong-key windows-license-with-key)]);
-        die("The provided product key didn't work...") if (match_has_tag('windows-wrong-key'));
-    }
-    else {
-        assert_and_click 'windows-no-prod-key';
-        assert_screen 'windows-select-system';
-        send_key_until_needlematch('windows-10-pro', 'down');
-        send_key 'alt-n';    # select OS (Win 10 Pro)
-        assert_screen 'windows-license';
+    if (get_var('WIN_UNATTENDED', '0')) {
+        # This test works onlywith CDMODEL=ide-cd due to windows missing scsi drivers which are installed via scsi iso
+        assert_screen 'windows-setup', 300;
+        send_key 'alt-n';    # next
+        prepare_win11 if (check_var('WIN_VERSION', '11'));
+
+        save_screenshot;
+        send_key 'alt-i';    # install Now
+        save_screenshot;
+        send_key 'alt-n';    # next
+        assert_screen 'windows-activate';
+        if (my $key = get_var('_SECRET_WINDOWS_10_PRO_KEY')) {
+            type_password $key . "\n";
+            assert_screen([qw(windows-wrong-key windows-license-with-key)]);
+            die("The provided product key didn't work...") if (match_has_tag('windows-wrong-key'));
+        }
+        else {
+            assert_and_click 'windows-no-prod-key';
+            assert_screen 'windows-select-system';
+            send_key_until_needlematch('windows-10-pro', 'down');
+            send_key 'alt-n';    # select OS (Win 10 Pro)
+            assert_screen 'windows-license';
+        }
+        send_key 'alt-a';    # accept eula
+        send_key 'alt-n';    # next
+        assert_screen 'windows-installation-type';
+        send_key 'alt-c';    # custom
+        assert_screen 'windows-disk-partitioning';
+        send_key 'alt-l';    # load driver
+        assert_screen 'windows-load-driver';
+        send_key 'alt-b';    # browse button
+        send_key 'c';
+        save_screenshot;
+        send_key 'c';    # go to second CD drive with drivers
+        send_key 'right';    # ok
+        sleep 0.5;
```

Comment:
Is this needed? If so, is sleeping for 500ms really enough, even if the workers are under heavy load?

Reply:
That has always been there and in fact the "good practices guide" recommends not to use `sleep` :man_shrugging:

Reply:
I'm not against using `sleep` here, but such a short sleep appears a bit ... short.

But ok, if this was working so far, let's keep it.